api/proto/teleport/legacy/types/types.proto

```diff
@@ -6914,6 +6919,39 @@ message PluginAWSICSettings {
   // GroupSyncLabelsFilter is used to specify filters that determine which AWS groups
   // should be included during synchronization.
   repeated AWSICResourceFilter group_sync_filters = 10 [(gogoproto.jsontag) = "group_sync_filters,omitempty"];
+
+  // Credentials represents the AWS credentials used by the Identity Center
+  // integration
+  AWSICCredentials credentials = 11 [(gogoproto.jsontag) = "credentials,omitempty"];
+}
+
+// AWSICCredentials holds the credentials for authenticating with AWS
+message AWSICCredentials {
+  option (gogoproto.equal) = true;
+  oneof source {
+    AWSICCredentialSourceSystem system = 1;
+    AWSICCredentialSourceOIDC oidc = 2;
+  }
+}
+
+// AWSICCredentialSourceSystem holds AWSIC credentials drawn from the ambient
+// system configuration
+message AWSICCredentialSourceSystem {
+  option (gogoproto.equal) = true;
+
+  // AwsRoleArn is an optional AWS role for the IC client to assume, overriding
+  // any roles
+  string aws_role_arn = 1 [(gogoproto.jsontag) = "aws_role_arn,omitempty"];
```

Comment:
Just for consistency https://github.com/gravitational/teleport/blob/master/api/proto/teleport/legacy/types/types.proto#L457 
```suggestion
  string AssumeRoleARN = 1 [(gogoproto.jsontag) = "assume_role_arn,omitempty"];
```